sydent/db/threepid_associations.py

```diff
@@ -95,6 +95,41 @@ def getMxid(self, medium, address):
 
         return row[0]
 
+    def getMxids(self, threepid_tuples):
+        cur = self.sydent.db.cursor()
+
+        cur.execute("CREATE TEMPORARY TABLE tmp_getmxids (medium VARCHAR(16), address VARCHAR(256))");
+
+        inserted_cap = 0
+        while inserted_cap < len(threepid_tuples):
+            cur.executemany(
+                "INSERT INTO tmp_getmxids (medium, address) VALUES (?, ?)",
+                threepid_tuples[inserted_cap:inserted_cap + 500]
+            )
+            inserted_cap += 500
+
+        res = cur.execute(
+            "SELECT gte.medium, gte.address, gte.ts, gte.mxid FROM global_threepid_associations gte "
+            "JOIN tmp_getmxids ON gte.medium = tmp_getmxids.medium AND gte.address = tmp_getmxids.address "
+            "WHERE gte.notBefore < ? AND gte.notAfter > ? "
+            "ORDER BY gte.ts DESC",
+            (time_msec(), time_msec())
```

Comment:
this is a pretty confusing query - "where the association is notBefore now and notAfter now"?  comment would be helpful, as it looks almost like it might be broken here...